clang/include/clang/Sema/Sema.h

```diff
@@ -3936,159 +3936,79 @@ class Sema final : public SemaBase {
 
   void redelayDiagnostics(sema::DelayedDiagnosticPool &pool);
 
-  void AddSYCLIntelBankBitsAttr(Decl *D, const AttributeCommonInfo &CI,
-                                Expr **Exprs, unsigned Size);
-  bool AnyWorkGroupSizesDiffer(const Expr *LHSXDim, const Expr *LHSYDim,
-                               const Expr *LHSZDim, const Expr *RHSXDim,
-                               const Expr *RHSYDim, const Expr *RHSZDim);
-  bool AllWorkGroupSizesSame(const Expr *LHSXDim, const Expr *LHSYDim,
-                             const Expr *LHSZDim, const Expr *RHSXDim,
-                             const Expr *RHSYDim, const Expr *RHSZDim);
-  void AddSYCLWorkGroupSizeHintAttr(Decl *D, const AttributeCommonInfo &CI,
-                                    Expr *XDim, Expr *YDim, Expr *ZDim);
   SYCLWorkGroupSizeHintAttr *
   MergeSYCLWorkGroupSizeHintAttr(Decl *D, const SYCLWorkGroupSizeHintAttr &A);
```

Comment:
Is there a reason why you haven't moved the Merge* functions?

Reply:
Good question. It's mostly because the `Merge*` functions start to pull other functions from `SemaDeclattr.cpp` to `SemaSYCL`, as well as changing up a certain part of `SemaDeclattr.cpp` and at point the git diff just loses all grasp on reality. I think it's mostly a git bug/problem: there's a tiny amount of changes required amongst the thousands of lines of deletion, and it starts to show whole functions as being "moved" across `SemaDeclattr.cpp` when in fact it's just surrounding code being deleted. It's like git just loses track of the context or something. Maybe a better diff heuristic would work better but if the default is unreadable, that's what everyone's going to see.

So given all that, and how difficult it would be to navigate the diff and git history, it's better to split it up into two patches. I have a follow-up patch which moves all of the `Merge*` and the rest of the `handle*` functions. The diff across both patches is unreadable: the diff of each patch individually is perfectly clear.

Reply:
Thanks for explaining. I'm alright with splitting the changes as you see fit.